testing.py

```diff
@@ -313,14 +315,14 @@ def __init__(self):
         self.requests = []
 
     def queue_response(self, response_code, media_type="text/html",
-                       other_headers=None, content=''):
+                       other_headers=None, content='', links=None):
         headers = {}
         if media_type:
             headers["Content-Type"] = media_type
         if other_headers:
             for k, v in other_headers.items():
                 headers[k.lower()] = v
-        self.responses.append(DummyHTTPResponse(response_code, headers, content))
+        self.responses.insert(0, DummyHTTPResponse(response_code, headers, content, links))
```

Comment:
This makes `queue_response` act like a queue instead of a stack, which is more accurate to the name and which makes the tests easier to read.